gosling/api.py

```diff
@@ -50,6 +50,52 @@ def display(self):
         display(self)
 
 
+def horizontal(*views, **kwargs):
+    return View(arrangement="horizontal", views=views, **kwargs)
+
+
+def vertical(*views, **kwargs):
+    return View(arrangement="vertical", views=views, **kwargs)
+
+
+def parallel(*views, **kwargs):
+    return View(arrangement="parallel", views=views, **kwargs)
+
+
+def serial(*views, **kwargs):
+    return View(arrangement="serial", views=views, **kwargs)
+
+
+# View utilities
+
+
+def overlay(*tracks, **kwargs):
+    """Compose an overlaid view from multiple tracks or overliad tracks."""
+    # Overlay requires a `width` and `height`. Check if provided as kwargs, otherwise
```

Comment:
Clarification question: I thought `width` and `height` are required properties in `SingleTrack` (which I think is the explicit type used here), but when I test with the current version, the package does not complain about the un-defined `width` and/or `height`. Am I missing something?:

```py
bar_base = gos.Track(data=mv).mark_bar().encode(
    x='start:G',
    xe='end:G',
    y='peak:Q',
    color=gos.Channel('sample:N', range=['steelblue', 'salmon', 'steelblue', 'salmon']),
).properties(width=100).view() # height is not specified, but this works
```

Reply:
Great question. This was a stylistic decisions on my part. 

https://github.com/manzt/ipygosling/blob/5c9e3bcee09aa5b1fbe928caee5210a075fc3cae/gosling/api.py#L53-L60

The constructor for `gosling.schema.core.Track` cannot be invoked with missing `width` or `height` since they are required properties. I overrode the signature in the top-level class above, so that there are defaults if not provided. I liked this pattern (similar to Altair), where you bind the `data` to the object and then can override the defaults with `.properties`.

```python
track = gos.Track(data)
gos.stack(track.properties(width=50, height=50), track.properties(width=50, height=100))

# vs
track = gos.Track(data, width=50, height=100) # will error if not provided
```

Reply:
This logic will also be removed if gosling.js a default setting.

Reply:
Ah, that makes sense! We can remove this part once Gosling.js is updated, but I think this shouldn't stop merging the PR to the main branch.